### PR TITLE
Change Bloom filter feature flag semantics; add telemetry

### PR DESCRIPTION
**Pull request by wkaluza**

### CODE REVIEW

File: libs/ledger/src/chain/main_chain.cpp [hunk 1]
```diff
@@ -1482,39 +1493,52 @@ DigestSet MainChain::DetectDuplicateTransactions(BlockHash const &starting_hash,
...
-  }
+
+    return duplicates;
+  };
+
+  DigestSet const duplicates =
+      search_chain_for_duplicates(enable_bloom_filter_ ? potential_duplicates : transactions);
 
   auto const false_positives =
       static_cast<std::size_t>(potential_duplicates.size() - duplicates.size());
```
Comment (ejfitzgerald) [inline libs/ledger/src/chain/main_chain.cpp]: I suspect this static_cast is redundant

**Review by ejfitzgerald — COMMENTED** [review]
Looks good one minor niggle

File: libs/ledger/src/chain/main_chain.cpp (hunk 1 above)
Comment (wkaluza) [reply libs/ledger/src/chain/main_chain.cpp]: Quite right.

**Review by ejfitzgerald — APPROVED** [review]
LGTM

**Pull request merged**